crates/sui-types/src/temporary_store.rs

```diff
@@ -1103,28 +1070,6 @@ impl<'backing> TemporaryStore<'backing> {
             );
             }
         }
-
-        // all SUI in storage rebate fields of input objects should flow either to the transaction storage rebate, or the non-refundable
-        // storage rebate pool
-        if total_input_rebate != gas_summary.storage_rebate + gas_summary.non_refundable_storage_fee
-        {
-            // TODO: re-enable once we fix the edge case with OOG, gas smashing, and storage rebate
-            /*return Err(ExecutionError::invariant_violation(
-                format!("SUI conservation failed--{} SUI in storage rebate field of input objects, {} SUI in tx storage rebate or tx non-refundable storage rebate",
-                total_input_rebate,
-                gas_summary.non_refundable_storage_fee))
-            );*/
-        }
-
-        // all SUI charged for storage should flow into the storage rebate field of some output object
-        if gas_summary.storage_cost != total_output_rebate {
-            // TODO: re-enable once we fix the edge case with OOG, gas smashing, and storage rebate
-            /*return Err(ExecutionError::invariant_violation(
-                format!("SUI conservation failed--{} SUI charged for storage, {} SUI in storage rebate field of output objects",
-                gas_summary.storage_cost,
-                total_output_rebate))
-            );*/
-        }
```

Comment:
this code can be safely removed as the behavior of an OOG for storage charges is not going to change and the logic was assuming that cost and rebate were always charged correctly. But in an OOG during storage charges we cannot perform proper operations on object storage cost and we still have to complete the transaction.
Conservation checks that have run since genesis proved consistency so those check can be safely removed

Reply:
I understand that in edge cases these checks don't hold.
But is there any invariant that we could still check, regarding storage rebates?